auth.go

```diff
@@ -8,6 +8,14 @@ import (
 	"github.com/pkg/errors"
 )
 
+// metadata holds additional information about Vault secrets (kv v2)
+type metadata struct {
```

Comment:
Given that `metadata` is used on `Secret`, a public struct, we should also make `metadata` a public struct.